src/tests/cli_tests.py

```diff
@@ -1018,6 +1018,46 @@ def test_encrypt_P3072_3072(self):
         cmd = EncryptElgamal.GPG_GENERATE_DSA_ELGAMAL_PATERN.format(3072, 3072, self.rnp.userid)
         self._encrypt_decrypt(self.gpg, self.rnp, 1, cmd)
 
+class EncryptEcdh(Encrypt):
+
+    GPG_GENERATE_ECDH_ECDSA_PATERN = """
+        Key-Type: ecdsa
+        Key-Curve: {0}
+        Key-Usage: sign auth
+        Subkey-Type: ecdh
+        Subkey-Usage: encrypt
+        Subkey-Curve: {0}
+        Name-Real: Test Testovich
+        Expire-Date: 1y
+        Preferences: twofish sha256 sha384 sha512 sha1 zlib
+        Name-Email: {1}"""
+
+    RNP_GENERATE_ECDH_ECDSA_PATERN = "18\n{0}\n"
+
+    def test_encrypt_nistP256(self):
+        cmd = EncryptEcdh.GPG_GENERATE_ECDH_ECDSA_PATERN.format("nistp256", self.rnp.userid)
+        self._encrypt_decrypt(self.gpg, self.rnp, 1, cmd)
```

Comment:
I guess you'll want to drop the key_id param on all these after your changes in fd15cc87e58af2c6273d34b51359e21c1eea7cb4.

Reply:
Indeed, this needs to be merged with "new approach" now. I'll do it just after ECDH fix is merged. So please don't merge until done